python_modules/dagster/dagster/_core/definitions/assets/definition/computation.py

```diff
@@ -82,12 +83,13 @@ def from_fn(
         output_mappings = {effect.spec.key.to_python_identifier(): effect for effect in effects}
 
         config_type = get_config_param_type(fn)
+        annotation_type = _extract_output_annotation_type_from_computation_fn(fn)
         op_def = OpDefinition(
             compute_fn=wrapped_fn,
             name=op_spec.name or wrapped_fn.name,
             ins={named_in.input_name: named_in.input for named_in in named_ins.values()},
             outs={
-                output_name: effect.to_out(can_subset=can_subset)
+                output_name: effect.to_out(can_subset=can_subset, annotation_type=annotation_type)
```

Comment:
[1] Should we have some additional validation that any asset which specifies its dagster type (not via annotation) is a subset of `annotation_type`, if present?

e.g.

`dg.AssetSpec(...).dagster_type(int)` but the return annotation is `Iterable[AssetResult[str]]`

Reply:
🙂‍↕️